src/site/_drafts/beautiful-efficient-images-for-e-commerce/index.md

```diff
@@ -0,0 +1,404 @@
+    products. (The [pinch-zoom](https://github.com/GoogleChromeLabs/pinch-zoom)
+    web component does this extremely well.) Don't just add a magnifying
+    function that simply increases the display size of a small image! You need
+    to provide high quality images with larger pixel dimensions, that display
+    well at larger sizes. Customers will probably accept a short delay while a
+    larger image loads. It can also be worth adding close-up images.
+
+    <br>
+
++   **Image carousels and 'more images'**
+
+    There are [many different ways to implement product carousels](https://www.hongkiat.com/blog/image-carousels-tips/)
+    and [mixed opinions](https://vwo.com/blog/image-slider-alternatives/) about
+    their value. Whatever you do, make sure to use lazy loading (see above) to
+    ensure you don't load offscreen images until you need to.
+
+## Optimize for search
+
++   Remember that people may navigate to your site via image search, so
+    your images need to stand out on their own. Images need to work well away
+    from the context of your website.
++   Images and SEO: find out more from
+    [Google image best practices](https://support.google.com/webmasters/answer/114016),
+    and use analytics to check what your customers search for.
+
+## Test, analyse, monitor, fix — repeat
+
+Make sure to include image performance testing in your workflow:
+
++   Use browser tools to manually check for bloated images: view the
+    Network panel and order by size. You can also use
+    [Lighthouse](https://developers.google.com/web/tools/lighthouse/) and other
+    tools to warn of large image files.
++   Monitor your site for load performance glitches, for example by using
+    the
+    [Google Analytics site speed](https://support.google.com/analytics/answer/1205784?hl=en)
+    feature.
++   Monitor payload sizes. You can use the [Lighthouse
+    API](https://developers.google.com/web/tools/lighthouse/) to continually
+    report on performance and warn for problems.
++   Compare the performance of your site against competitors using tools
+    such as [Test My
+    Site](https://www.thinkwithgoogle.com/intl/en-gb/feature/testmysite) and
+    the
+    [Chrome User Experience Report](https://developers.google.com/web/tools/chrome-user-experience-report/).
++   Consider adding warnings to your build process, for example if
+    developers attempt to commit oversized files.
++   Sounds obvious, but if you're monitoring image usage on your live site,
+    make sure to test category and product pages — not just the home page!
+
+## Next steps
+
++   [web.dev/fast](https://web.dev/fast#topic-Optimize-your-images) has a
+    range of guides and codelabs to help you optimize your images.
++   Our codelab
+    [Optimize images for e-commerce](https://docs.google.com/document/d/1m7xWYKf7gJ4IeLkKKc7FfhFtk0vZi0BM2nxJhzDPwHU/edit#heading=h.cgvdu9aeouaf)
+    [TODO: web.dev link] shows how to create beautiful images for online
```

Comment:
Missing TODO

Reply:
I'll removed this temporarily, until the codelab is done.